tests/phpunit/integration/Core/Authentication/Setup_Test.php

```diff
@@ -206,7 +207,73 @@ function( $preempt, $args, $url ) use ( $context, &$proxy_server_requests, $has_
 		} catch ( WPDieException $exception ) {
 			$error = $has_credentials ? 'Test error message.' : 'test_error_code';
 			$this->assertStringContainsString(
-				sprintf( 'The request to the authentication proxy has failed with an error: %s', $error ),
+				sprintf(
+					'The request to the authentication proxy has failed with an error: %1$s %2$s.',
+					$error,
+					$oauth_proxy_failed_help_link
+				),
+				$exception->getMessage()
+			);
+		}
+
+		$this->assertCount( 1, $proxy_server_requests );
+	}
+
+	/**
+	 * @dataProvider data_conditionally_syncs_site_fields
+	 */
+	public function test_handle_action_setup_start__invalid_url( $has_credentials ) {
+		$user_id = $this->factory()->user->create( array( 'role' => 'administrator' ) );
+		wp_set_current_user( $user_id );
+
+		$context                      = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE, new MutableInput() );
+		$setup                        = new Setup( $context, new User_Options( $context ), new Authentication( $context ) );
+		$oauth_proxy_failed_help_link = $setup->get_oauth_proxy_failed_help_link();
+		$setup->register();
+
+		if ( $has_credentials ) {
+			$this->fake_proxy_site_connection();
+		}
+
+		$_GET['code']  = 'test-code';
+		$_GET['nonce'] = wp_create_nonce( Google_Proxy::ACTION_SETUP_START );
+
+		$proxy_server_requests = array();
+
+		add_filter(
+			'pre_http_request',
+			function( $preempt, $args, $url ) use ( $context, &$proxy_server_requests, $has_credentials ) {
```

Comment:
`$has_credentials` is unused within the function

```suggestion
			function( $preempt, $args, $url ) use ( $context, &$proxy_server_requests ) {
```